### PR TITLE
langchain[minor]: Added Search Functionality to FireCrawl Document Loader

### PR DESCRIPTION
**Pull request by rafaelsideguide**
This PR adds search functionality to the FireCrawl document loader, enabling users to search and retrieve specific data from the web.

This enhancement builds on the original FireCrawl loader introduced in PR [#5180](https://github.com/langchain-ai/langchainjs/pull/5180).

Thank you!

### CODE REVIEW

**Review by bracesproul — CHANGES_REQUESTED** [review]
Thanks for pushing this up! I left a couple comments, and have one bigger request:
This search functionality should be a retriever, and not a document loader. Would you be willing to instead refactor this PR to not add any new functionality to the document loader, but instead create a new firecrawl retriever which has the ability to search?

You can use [this retriever](https://github.com/langchain-ai/langchainjs/blob/main/libs/langchain-exa/src/retrievers.ts) for reference

**Comment by rafaelsideguide** [comment]
> Thanks for pushing this up! I left a couple comments, and have one bigger request: This search functionality should be a retriever, and not a document loader. Would you be willing to instead refactor this PR to not add any new functionality to the document loader, but instead create a new firecrawl retriever which has the ability to search?
> 
> You can use [this retriever](https://github.com/langchain-ai/langchainjs/blob/main/libs/langchain-exa/src/retrievers.ts) for reference

@bracesproul I just changed the search implementation following your suggestions!

**Comment by rafaelsideguide** [comment]
Hey @bracesproul I've made the changes you asked for in the firecrawl retriever. Could you take a look when you get a chance? I want to make sure it's all good.

File: langchain/src/document_loaders/web/firecrawl.ts [hunk 1]
```diff
@@ -113,4 +113,4 @@ export class FireCrawlLoader extends BaseDocumentLoader {
         })
     );
   }
-}
+}
```
Comment (bracesproul) [inline langchain/src/document_loaders/web/firecrawl.ts:116]:
```suggestion
}

```

**Pull request closed without merging**